typing/typedtree.mli

```diff
@@ -986,15 +992,18 @@ and open_declaration = module_expr open_infos
 
 and include_kind =
   | Tincl_structure
-  | Tincl_functor of (Ident.t * module_coercion) list
+  | Tincl_functor of
+      (Ident.t * module_coercion) list * Types.module_representation
       (* S1 -> S2 *)
-  | Tincl_gen_functor of (Ident.t * module_coercion) list
+  | Tincl_gen_functor of
+      (Ident.t * module_coercion) list * Types.module_representation
       (* S1 -> () -> S2 *)
 
 and 'a include_infos =
     {
      incl_mod: 'a;
      incl_type: Types.signature;
+     incl_repr: Types.module_representation;
```

Comment:
Reading this, I wondered why there is n `module_representation` in the functor cases above, but not the structure case. And then also was then surprised that there is also a module representation here. Is this redundant with the ones that exist in the functor cases?

(I suppose not, maybe the ones in the functor cases are for the input module, or something, and I'll understand that better as I read the rest of the PR. But probably this should be made clear here with a comment or by changing `include_kind` to use inline records with clarifying field names.)